### PR TITLE
Update tutorials and switch to using docstrings

### LINKED ISSUE #68: [Documentation] Docstring for gsa function missing
It would be nice to add docstring for gsa function and include the argument lists for different methods.

Currently, the docs are like:

### API

`function gsa(f, method::Morris, p_range::AbstractVector; batch=false, kwargs...)`

Something like [this](https://clima.github.io/OceananigansDocumentation/stable/appendix/library/#Models) would be better.

[openjournals/joss-reviews#4561](https://github.com/openjournals/joss-reviews/issues/4561)

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
Fixes #68 and #69

### CODE REVIEW

File: docs/src/methods/easi.md [hunk 1]
```diff
@@ -1,17 +1,9 @@
...
+```@docs
+EASI(; max_harmonic::Int = 10, dct_method::Bool = false)
 ```
 
-`EASI` has the following keyword arguments:
-
-- `max_harmonic`: Maximum harmonic of the input frequency for which the output power spectrum is analyzed for. Defaults to `10`.
-- `dct_method`: Use Discrete Cosine Transform method to compute the power spectrum. Defaults to `false`.
-
 ## Method Details
```
Comment (ChrisRackauckas) [inline docs/src/methods/easi.md:7]: This stuff should also be moved to the docstring

File: docs/src/methods/easi.md [hunk 2]
```diff
@@ -26,8 +18,8 @@ available input samples.
 
 ### API
 
-```julia
-function gsa(f, method::EASI, p_range; N, batch = false, rng::AbstractRNG = Random.default_rng(), kwargs...)
+```@docs
+gsa(f, method::EASI, p_range; samples, batch = false, rng::AbstractRNG = Random.default_rng(), kwargs...)
 ```
 
 ### Example
```
Comment (ChrisRackauckas) [inline docs/src/methods/easi.md:25]: also for the docstring

**Comment by ChrisRackauckas** [comment]
squash out the .DS_Stores ....

File: src/GlobalSensitivity.jl [hunk 3]
```diff
@@ -17,8 +17,47 @@ include("easi_sensitivity.jl")
...
+
+```julia
+effects = gsa(f, method, A, B; batch=false)
+```
+
+where `A` and `B` are design matrices with each row being a set of parameters. Note that `generate_design_matrices`
+from [QuasiMonteCarlo.jl](https://github.com/JuliaDiffEq/QuasiMonteCarlo.jl) can be used to generate the design
+matrices.
+"""
+gsa(f, method, param_range; samples, batch = false)
```
Comment (ChrisRackauckas) [inline src/GlobalSensitivity.jl:56]: This is a function call, not a function definition.

**Comment by ChrisRackauckas** [comment]
This is a bit odd. Most people would never see the docstrings listed like this. Instead, it should just have a single

```julia
function gsa end
```

which has a high level docstring that defines the interface. That interface should have a `::AbstractGlobalSensitivityAlgorithm` that then says to see those for more information on the specific algorithms, and then each algorithm should have a docstring on the struct (which would then be the one that is caught in `?EASI`, instead of on the constructor function).

That would then make `?gsa` give the high level information and `?Sobol` give the details for `gsa` on `Sobol` calls.

File: src/GlobalSensitivity.jl (hunk 3 above)
Comment (Vaibhavdixit02) [reply src/GlobalSensitivity.jl]: Updated it, but that's not wrong btw https://docs.julialang.org/en/v1/manual/documentation/#Functions-and-Methods-2 the second snippet in this section has an example of doing that

**Comment by Vaibhavdixit02** [comment]
I preferred having the docstring on the constructor over the struct because that's more useful to the user since they have to use it, looking at the struct doesn't give insight into how to construct the object

`?gsa` currently gives docstrings for all methods and the general interface, I agree it's a bit cumbersome but then that's how most functions with multiple methods show the docstring across julia. We could move the specific method's docstrings into the constructor docstring but I am not sure if that would be sensible since that lessens the discoverability?

**Comment by ChrisRackauckas** [comment]
> I preferred having the docstring on the constructor over the struct because that's more useful to the user since they have to use it, looking at the struct doesn't give insight into how to construct the object

No, the docstring should be about the constructor yes. But if you put the docstring about the constructor on the struct then the user will more easily be able to see it.

> ?gsa currently gives docstrings for all methods and the general interface, I agree it's a bit cumbersome but then that's how most functions with multiple methods show the docstring across julia. We could move the specific method's docstrings into the constructor docstring but I am not sure if that would be sensible since that lessens the discoverability?

That's just not useful. If `?gsa` documents the interface and then `?alg` gives specific details about the algorithm, then it's easily discoverable and simpler to read.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
Just format and I think we're good to go.

**Comment by Vaibhavdixit02** [comment]
I don't know what's up with this format job, can't seem to find anything locally 🤔

**Pull request merged**